src/Workspaces/Remote/ServiceHub/Services/SnapshotService.JsonRpcAssetSource.cs

```diff
@@ -44,7 +44,7 @@ public override async Task<IList<(Checksum, object)>> RequestAssetsAsync(int sco
                                 (s, c) => ReadAssets(s, scopeId, checksums, serializerService, c), cancellationToken);
                         }, cancellationToken).ConfigureAwait(false);
                     }
-                    catch (Exception ex) when (ReportUnlessCanceled(ex, cancellationToken))
```

Comment:
Does this imply that we're some how mixing up cancellation tokens and getting OperationCanceledExceptions when we didn't expect them?

Reply:
This might be the case if we were checking to see if the cancellation token provided in the exception was the same as the cancellation token seen here. However, in this case we're doing a more general check of whether cancellation was requested at all. In other words, we were expecting this:

* If cancellation is requested, any `OperationCanceledException` is acceptable regardless of if/which token is indicated in the exception.
* If cancellation is not requested at all, `OperationCanceledException` is not expected to occur.

Reply:
>This pull request doesn't fix the underlying problem, but rather acknowledges that we have observed the problem 

That's correct. My intention was to avoid more Watson hits with the same root cause. It would be useful to link an issue that tracks the underlying cause if such issue exists, or create a new one. Do we have one already?

Reply:
Filed https://github.com/dotnet/roslyn/issues/39723